src/Nerdbank.Streams/MultiplexingStream.Channel.cs

```diff
@@ -290,6 +290,20 @@ public PipeWriter Output
             /// </summary>
             internal Task OptionsApplied => this.optionsAppliedTaskSource?.Task ?? Task.CompletedTask;
 
+            /// <summary>
+            /// Gets the exception due to which this channel is going to be disposed, if there is one.
+            /// </summary>
+            internal Exception? ShutdownException
+            {
+                get
+                {
+                    lock (this.SyncObject)
```

Comment:
The reason I did this was that we always acquire a lock before updating a shutdown exception. Thus, I though that we should be safe and acquire a lock before the reading this field so that the field doesn't get updated while we are trying to read it.

Reply:
> so that the field doesn't get updated while we are trying to read it.

It sounds like you're concerned about "torn state". Given this field is the size of a pointer, we can never read torn state. We may read the old value or the new one, which is no different than if we had a lock around it.
Using a lock adds memory fences, and I believe it will help avoid issues with per-core CPU caches so that we're sure to actually read the new value if it was set by another thread, so there _can_ still be value with locking, but TBH that's an advanced area that I'm not super familiar with, and so far haven't ever identified a place where not having a lock caused a regression.